.github/workflows/engine-and-cli.yml

```diff
@@ -22,6 +22,13 @@ permissions:
 
 jobs:
   test:
+    uses: ./.github/workflows/_hack_make.yml
+    secrets: inherit
+    with:
+      mage-targets: engine:testrace
+      size: "dagger-v0-11-4-16c-nogc"
```

Comment:
I would leave the default name as it was, and instead introduce a new job with `-nogc` appended. This way, it will be clearer that we are introducing a new variant. It's easier to explain & understand, rather than what we had now becomes `baseline`, and the default changed to not having GC.